kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveSessionParams.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.api.stores;
+
+import static dev.responsive.kafka.internal.utils.StoreUtil.durationToMillis;
+
+import dev.responsive.kafka.internal.stores.SchemaTypes.SessionSchema;
+import dev.responsive.kafka.internal.utils.TableName;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ResponsiveSessionParams {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ResponsiveSessionParams.class);
+
+  // TODO: should we enforce a real upper bound on the number of segments/partitions that are
+  //  created? Is there a maximum recommended for Scylla, or for this compaction strategy, etc?
+  private static final long MINIMUM_ALLOWED_NUM_SEGMENTS = 2L;
+  private static final long MAXIMUM_ALLOWED_NUM_SEGMENTS = Long.MAX_VALUE;
+  private static final long MAXIMUM_DEFAULT_NUM_SEGMENTS = 100L;
+  private static final long DEFAULT_NUM_SEGMENTS = 10L; // space amplification = 1.1x
+
+  private final TableName name;
+  private final SessionSchema schemaType;
+  private final long inactivityGapMs;
+  private final long gracePeriodMs;
+  private final boolean retainDuplicates;
```

Comment:
Might have forgotten to mention this before --there is no "retain duplicates" for session stores. It's a special parameter of the WindowStore specifically. One could argue it's a bit of a hack, but basically the WindowStore gets reused by Kafka Streams for stream-stream joins, where it effectively treats the WindowStore as a regular KV-plus-timestamp store. But it has to be able to hold multple values for the same key-timestamp since you could have multiple events with the same key and timestamp in your join, thus Streams will set `retainDuplicates` to true for WindowStores that are to be used for stream-stream joins (as opposed to aggregations, which should have strict update semantics)

tl;dr we should remove all the retainDuplicates stuff for session stores